File: 단계별로 풀어보기/기본 수학2/1978/소수 찾기.py
import sys

def init():
    N = int(sys.stdin.readline())
    nums = list(map(int,sys.stdin.readline().split()))
    cnt = 0
    
    for i in range(N):
        prime = True
        j = 2
        
        if nums[i] == 1:
            continue
        
        while j < nums[i] and prime:
            if nums[i]%j == 0:
                prime = False
                break
            j+=1
        
        if prime:
            cnt+=1
            
    print(cnt)
    
    return 0

init()

File: 단계별로 풀어보기/기본 수학2/1978/소수 찾기.cpp
#include <iostream>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    int cnt=0; //초기화 필요
    cin >> N;
    int* nums = new int[N];
    
    for(int i = 0; i < N; i++)
        cin >> nums[i];
    
    for(int i = 0; i < N; i++){
        bool prime = true;
        
        if(nums[i]==1)
            continue;
        
        for(int j = 2; j < nums[i]; j++){
            if(nums[i]%j==0){
                prime = false;
                break;
            }
        }
        
        if(prime)
            cnt++;
    }
    
    delete nums;
    cout << cnt;
    return 0;
}
